python/migrate.py

```diff
@@ -1,102 +1,235 @@
 import json
-from typing import Any, Dict, List
-
 import mgp
 import mysql.connector as mysql_connector
 import oracledb
 import pyodbc
+import threading
+
+from typing import Dict, Any, List
+
+
+class Constants:
+    I_COLUMN_NAME = 0
+    CURSOR = "cursor"
+    COLUMN_NAMES = "column_names"
+    CONNECTION = "connection"
+    BATCH_SIZE = 1000
+
+
+def _query_is_table(table_or_sql: str) -> bool:
+    return len(table_or_sql.split()) == 1
+
+
+def _load_config(path: str) -> Dict[str, Any]:
+    try:
+        with open(path, mode="r") as config:
+            return json.load(config)
+    except Exception:
+        raise OSError("Could not open/read file.")
 
-I_COLUMN_NAME = 0
+
+def _combine_config(config: mgp.Map, config_path: str) -> Dict[str, Any]:
+    assert len(config_path), "Path must not be empty"
+    config_items = _load_config(path=config_path)
+    print(config, config_items)
+
+    for key, value in config_items.items():
+        config[key] = value
+    print(config, config_items)
+    return config
+
+
+def _name_row_cells(row_cells, column_names) -> Dict[str, Any]:
+    return dict(map(lambda column, value: (column, value), column_names, row_cells))
+
+
+def _check_params_type(params: Any) -> None:
+    if not not isinstance(params, (dict, list, tuple)):
+        raise TypeError(
+            "MySQL query parameter values must be passed in a container of type List[Any] or Map."
+        )
+
+
+##### MYSQL
+
+mysql_dict = {}
+
+
+def init_migrate_mysql(
+    table_or_sql: str,
+    config: mgp.Map,
+    config_path: str = "",
+    params: mgp.Nullable[mgp.Any] = None,
+):
+    global mysql_dict
+
+    if params:
+        _check_params_type(params)
+    print(config, config_path)
+    if len(config_path) > 0:
+        config = _combine_config(config=config, config_path=config_path)
+
+    if _query_is_table(table_or_sql):
+        table_or_sql = f"SELECT * FROM {table_or_sql};"
+
+    if threading.get_native_id not in mysql_dict:
+        mysql_dict[threading.get_native_id] = {}
+
+    if Constants.CURSOR not in mysql_dict[threading.get_native_id]:
+        mysql_dict[threading.get_native_id][Constants.CURSOR] = None
+
+    if mysql_dict[threading.get_native_id][Constants.CURSOR] is None:
+        connection = mysql_connector.connect(**config)
+        cursor = connection.cursor(buffered=True)
+        cursor.execute(table_or_sql, params=params)
+
+        mysql_dict[threading.get_native_id][Constants.CONNECTION] = connection
+        mysql_dict[threading.get_native_id][Constants.CURSOR] = cursor
+        mysql_dict[threading.get_native_id][Constants.COLUMN_NAMES] = [
+            column[Constants.I_COLUMN_NAME] for column in cursor.description
+        ]
 
 
-@mgp.read_proc
 def mysql(
```

Comment:
All parameters here are unused

Reply:
Yeah that's by design of batched procedures.